kbs/src/policy_engine/opa/mod.rs

```diff
@@ -32,33 +32,33 @@ impl PolicyEngineInterface for Opa {
         &self,
         resource_path: String,
         input_claims: String,
-    ) -> Result<bool, ResourcePolicyError> {
+    ) -> Result<bool, PolicyEngineError> {
         let mut engine = regorus::Engine::new();
 
         // Add policy as data
         engine
             .add_policy_from_file(self.policy_path.clone())
-            .map_err(|_| ResourcePolicyError::PolicyLoadError)?;
+            .map_err(|_| PolicyEngineError::PolicyLoadError)?;
 
         // Add resource path as data
         let resource_path_object =
             regorus::Value::from_json_str(&format!("{{\"resource-path\":\"{}\"}}", resource_path))
-                .map_err(|_| ResourcePolicyError::ResourcePathError)?;
+                .map_err(|_| PolicyEngineError::ResourcePathError)?;
 
         engine
             .add_data(resource_path_object)
-            .map_err(|_| ResourcePolicyError::DataLoadError)?;
+            .map_err(|_| PolicyEngineError::DataLoadError)?;
 
         // Add TCB claims as input
         engine
             .set_input_json(&input_claims)
-            .map_err(|_| ResourcePolicyError::InputError)?;
+            .map_err(|_| PolicyEngineError::InputError)?;
 
         let res = engine.eval_bool_query("data.policy.allow".to_string(), false)?;
         Ok(res)
     }
 
-    async fn set_policy(&mut self, policy: String) -> Result<(), ResourcePolicyError> {
+    async fn set_policy(&mut self, policy: String) -> Result<(), PolicyEngineError> {
```

Comment:
can we change `policy: &str` in this signature, so we don't have potentially clone on the caller site?

Reply:
Nice idea